modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -338,17 +348,47 @@ TensorMechanicsAction::actOutputGeneration()
     // Loop through output aux variables
     for (auto out : _generate_output)
     {
-      auto type = data[out]._auxkernel;
-      auto variable = data[out]._variable;
-
-      InputParameters params = _factory.getValidParams(type);
-      params.set<AuxVariableName>("variable") = variable;
-      params.set<MaterialPropertyName>("rank_two_tensor") = data[out]._tensor;
-      params.set<MultiMooseEnum>("execute_on") = "timestep_end";
-      params.set<unsigned int>("index_i") = data[out]._index.first;
-      params.set<unsigned int>("index_j") = data[out]._index.second;
-
-      _problem->addAuxKernel(type, variable, params);
+      std::string type = "";
+      InputParameters params = emptyInputParameters();
+
+      // RankTwoAux
+      for (const auto & r2a : _ranktwoaux_table)
+        for (unsigned int a = 0; a < 3; ++a)
+          for (unsigned int b = 0; b < 3; ++b)
+            if (r2a.first + '_' + _component_table[a] + _component_table[b] == out)
+            {
+              type = "RankTwoAux";
+              params = _factory.getValidParams(type);
+              params.set<MaterialPropertyName>("rank_two_tensor") = r2a.second;
+              params.set<unsigned int>("index_i") = a;
+              params.set<unsigned int>("index_j") = b;
+            }
+
+      // RankTwoScalarAux
+      for (const auto & r2sa : _ranktwoscalaraux_table)
+        for (const auto & t : r2sa.second.second)
+          if (r2sa.first + '_' + t == out)
+          {
+            const auto r2a = _ranktwoaux_table.find(t);
+            if (r2a != _ranktwoaux_table.end())
+            {
+              type = "RankTwoScalarAux";
+              params = _factory.getValidParams(type);
+              params.set<MaterialPropertyName>("rank_two_tensor") = r2a->second;
+              params.set<MooseEnum>("scalar_type") = r2sa.second.first;
+            }
+            else
+              mooseError("Internal error. The permitted tensor shortcuts in '_ranktwoscalaraux_table' must be keys in the '_ranktwoaux_table'.");
```

Comment:
Since this is an error that users are likely to encounter frequently, it would be good to dump out what all of the options are here rather than pointing them to something in the source code.